Kconfig

```diff
@@ -147,4 +147,32 @@ menu "LittleFS"
             In LittleFS flush() does not write data to the flash, and fsync() call needed after.
             With this feature fflush() will write data to the storage.
 
+    config LITTLEFS_OPEN_DIR
+        bool "Support opening directory"
+        default "n"
+        depends on !LITTLEFS_USE_ONLY_HASH && LITTLEFS_SPIFFS_COMPAT
```

Comment:
The API `open` can create directory, so this doesn't need to depend on `CONFIG_VFS_SUPPORT_DIR`.

Reply:
But does it make sense to create a directory if the virtual filesystem doesn't support it?

Reply:
Yes, it makes sense, because `open` can create directory, beside this, if the directory exists, users also need not create the directory.

Reply:
I don't see how `open` can create a directory if `vfs.mkdir_p` is NULL when `CONFIG_VFS_SUPPORT_DIR` is not defined.

Can we add a test for this, too?

Reply:
This feature depends on `LITTLEFS_SPIFFS_COMPAT`, so open can create directory, I have added test case [here](https://github.com/joltwallet/esp_littlefs/pull/85/files#diff-1579522f57beed6b5eac54f18dacaef9245df4c63dd14fb562b8785224f88603R1171).

Reply:
so I'm like 90% sure we'll need to have this depend on `CONFIG_VFS_SUPPORT_DIR`, and I also think that `LITTLEFS_SPIFFS_COMPAT` also needs to depend on `CONFIG_VFS_SUPPORT_DIR`. The unittests also need `CONFIG_VFS_SUPPORT_DIR` guards. Since I don't want to bloat this PR, I'll open that as a separate PR and I'll add you as a reviewer in the next day or two.